commands/webhook.go

```diff
@@ -0,0 +1,472 @@
+		}
+	} else {
+		teams = getTeamsFromTeamArgs(c, args)
+	}
+
+	for i, team := range teams {
+		if team == nil {
+			printer.PrintError("Unable to find team '" + args[i] + "'")
+			continue
+		}
+
+		// Fetch all hooks with a very large limit so we get them all.
+		incomingResult := make(chan StoreResult, 1)
+		go func() {
+			incomingHooks, response := c.GetIncomingWebhooksForTeam(team.Id, 0, 100000000, "")
+			incomingResult <- StoreResult{Data: incomingHooks, Err: response.Error}
+			close(incomingResult)
+		}()
+		outgoingResult := make(chan StoreResult, 1)
+		go func() {
+			outgoingHooks, response := c.GetOutgoingWebhooksForTeam(team.Id, 0, 100000000, "")
+			outgoingResult <- StoreResult{Data: outgoingHooks, Err: response.Error}
+			close(outgoingResult)
+		}()
+
+		if result := <-incomingResult; result.Err == nil {
+			printer.Print(fmt.Sprintf("Incoming webhooks for %s (%s):", team.DisplayName, team.Name))
+			hooks := result.Data.([]*model.IncomingWebhook)
+			for _, hook := range hooks {
+				printer.Print("\t" + hook.DisplayName + " (" + hook.Id + ")")
+			}
+		} else {
+			printer.PrintError("Unable to list incoming webhooks for '" + args[i] + "'")
+		}
+
+		if result := <-outgoingResult; result.Err == nil {
+			hooks := result.Data.([]*model.OutgoingWebhook)
+			printer.Print(fmt.Sprintf("Outgoing webhooks for %s (%s):", team.DisplayName, team.Name))
+			for _, hook := range hooks {
+				printer.Print("\t" + hook.DisplayName + " (" + hook.Id + ")")
+			}
+		} else {
+			printer.PrintError("Unable to list outgoing webhooks for '" + args[i] + "'")
+		}
+	}
+	return nil
+}
+
+func createIncomingWebhookCmdF(c client.Client, command *cobra.Command, args []string) error {
+	printer.SetSingle(true)
+
+	channelArg, errChannel := command.Flags().GetString("channel")
+	if errChannel != nil || channelArg == "" {
+		return errors.New("Channel is required")
+	}
+	channel := getChannelFromChannelArg(c, channelArg)
+	if channel == nil {
+		return errors.New("Unable to find channel '" + channelArg + "'")
+	}
+
+	userArg, errUser := command.Flags().GetString("user")
+	if errUser != nil || userArg == "" {
+		return errors.New("User is required")
+	}
+	user := getUserFromUserArg(c, userArg)
+	if user == nil {
+		return errors.New("Unable to find user '" + userArg + "'")
+	}
+
+	displayName, _ := command.Flags().GetString("display-name")
+	description, _ := command.Flags().GetString("description")
+	iconURL, _ := command.Flags().GetString("icon")
+	channelLocked, _ := command.Flags().GetBool("lock-to-channel")
+
+	incomingWebhook := &model.IncomingWebhook{
+		ChannelId:     channel.Id,
+		DisplayName:   displayName,
+		Description:   description,
+		IconURL:       iconURL,
+		ChannelLocked: channelLocked,
+		Username:      user.Username,
+	}
+
+	createdIncoming, respIncomingWebhook := c.CreateIncomingWebhook(incomingWebhook)
+	if respIncomingWebhook.Error != nil {
+		return respIncomingWebhook.Error
+	}
+
+	printer.Print("Id: " + createdIncoming.Id)
```

Comment:
To comply with the JSON output, we have to replace this with:

```suggestion
	tpl := `Id: {{.Id}}
Display Name: {{.DisplayName}}`

	printer.Print(tpl, createdIncoming)
```

and remove the following line